### PR TITLE
Bg_Dblue_Balance

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c [hunk 1]
```diff
@@ -32,56 +41,689 @@ const ActorInit Bg_Dblue_Balance_InitVars = {
...
+    } else {
+        this->unk_183 -= 6;
+    }
+}
+
+void func_80B8296C(GlobalContext* globalCtx, Vec3f* arg1, f32 arg2) {
+    s32 i;
+    Vec3f sp60;
+    f32 temp_f0;
+    f32 temp_f20 = arg2 * 2.000000000f;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:267]: Are all those zeroes required?
```suggestion
    f32 temp_f20 = arg2 * 2.0f;
```

Comment (Kelebek1) [reply src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:267]: Yeah they're required.

Comment (hensldm) [reply src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:267]: Checkedout the branch to do some testing, and it looks like `arg2 * 2` works.

File: src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c [hunk 2]
```diff
@@ -32,56 +41,689 @@ const ActorInit Bg_Dblue_Balance_InitVars = {
...
+
+    sp48.x = -138.28f;
+    sp48.y = 0.0f;
+    sp48.z = 0.0f;
+
+    Matrix_MultiplyVector3fByState(&sp48, &sp3C);
+
+    this->unk_15C = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_DBLUE_BALANCE, sp3C.x, sp3C.y, sp3C.z, 0,
+                                this->dyna.actor.shape.rot.y, 0, 0x100);
+    if (this->unk_15C == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:152]:
```suggestion
    if (this->unk_15C == NULL) {
```

File: src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c [hunk 3]
```diff
@@ -32,56 +41,689 @@ const ActorInit Bg_Dblue_Balance_InitVars = {
...
+    BgDblueBalance* this = THIS;
+    BgDblueBalanceStruct2* ptr2 = &D_80B83A20[BGDBLUEBALANCE_GET_300(&this->dyna.actor)];
+    Actor* sp38;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Balance/func_80B8259C.s")
+    Gfx_DrawDListOpa(globalCtx, ptr2->unk_00);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Balance/func_80B8264C.s")
+    if (!(BGDBLUEBALANCE_GET_300(&this->dyna.actor)) && (this->unk_160 != 0)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:653]:
```suggestion
    if (!(BGDBLUEBALANCE_GET_300(&this->dyna.actor)) && (this->unk_160 != NULL)) {
```

File: src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c [hunk 4]
```diff
@@ -32,56 +41,689 @@ const ActorInit Bg_Dblue_Balance_InitVars = {
...
+        balance = (BgDblueBalance*)this->unk_15C;
+        balance->unk_164 = &this->dyna.actor;
+    }
+
+    sp48.x = 138.28f;
+    Matrix_MultiplyVector3fByState(&sp48, &sp3C);
+
+    this->unk_160 = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_DBLUE_BALANCE, sp3C.x, sp3C.y, sp3C.z, 0,
+                                this->dyna.actor.shape.rot.y, 0, 0x200);
+    if (this->unk_160 == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c:164]:
```suggestion
    if (this->unk_160 == NULL) {
```

File: src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.h [hunk 5]
```diff
@@ -6,12 +6,43 @@
...
+    /* 0x10 */ f32 unk_10;
+} BgDblueBalanceStruct; // size = 0x14
 
 typedef struct BgDblueBalance {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x24];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x015C */ Actor* unk_15C;
+    /* 0x0160 */ Actor* unk_160;
+    /* 0x0164 */ Actor* unk_164;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.h:25]:
```suggestion
    /* 0x015C */ struct BgDblueBalance* unk_15C;
    /* 0x0160 */ struct BgDblueBalance* unk_160;
    /* 0x0164 */ struct BgDblueBalance* unk_164;
```
Looks like these are always `BgDblueBalance*` and access fields from it. Any reason not to just have the members be of that type?

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**